### ISSUE TITLE
KeyError in line 2 of /Natrix2/rules/pychop.smk: 'pychopper'  

### ISSUE THREAD

**Issue by timyerg** [dbeisser/Natrix2#3]
Hello!
Thank you for the tool. I am trying to run it on the server via conda for 16S Nanopore reads.
So, I installed the environment as in the tutorial, copied natrix2 folder, added raw reads in the folder named as the project and modified Nanopore_18S.yaml file for my needs (my files already assembled and demultiplexed).
When I start the job, it is killed after a minute with the message:

```
KeyError in line 2 of /Natrix2/rules/pychop.smk:
'pychopper'
  File "/Natrix2/Snakefile", line 57, in <module>
  File "/Natrix2/rules/pychop.smk", line 2, in <module>
```

I tried to tun it before and back then it was installing some missing packages and died after it. Since then I completely recreated new folder for the project and deleted environment and then reinstalled it. This time it is not trying to install missing packages, though I made sure to delete all old files. Can't figure out what is causing the error.

Here is the screenshot of units.tsv table:
![Screenshot from 2024-02-26 10-54-09](https://github.com/dbeisser/Natrix2/assets/41282261/82085744-d7f6-4eaf-ab3e-722be6ccd430)

My command:

```
conda activate natrix

cd $WS/Natrix2

python3 create_dataframe.py Nanopore16S.yaml 
snakemake --use-conda --configfile Nanopore16S.yaml --cores {THREADS}
```

Best,

**Comment by adeep619** [comment]
Hi there,
thanks for using this pipeline :)
could you please share the primer table file? Usually, this error occurs when the file names do not match probe column in the primer table CSV file.
thank you

**Comment by timyerg** [comment]
Thank you for the fast reply!
I don't have it since the primers were already removed. I thought that setting parameters for already assembled reads would override it. My bad.
I will create the table and try to run it again. 
Hopefully will close the issue after it...

**Comment by timyerg** [comment]
Still getting the same error.
Is there something I am missing?

I created a .csv file with the same name as the project and located it in the root of the natrix2 folder. Names are not in the same order as "units.tsv".

![Screenshot from 2024-02-26 10-54-09](https://github.com/dbeisser/Natrix2/assets/41282261/a2d89c4b-bd88-4635-9be9-f1ab763d35b3)

![Screenshot from 2024-02-26 11-52-00](https://github.com/dbeisser/Natrix2/assets/41282261/274edc17-f7ec-4998-8ea5-ce25e0d2e984)

**Comment by adeep619** [comment]
Hi there,
I apologize for the inconvenience caused. We have just made some updates to  Nanopore.yaml. This will enable you to turn off Pychopper for primer trimming and use other quality filtering parameters as well. Please use the dev branch for now. We plan to merge it with the main branch soon.
iIhope that will solve your problem. otherwise, feel free to write here.
Best,
Aman

**Comment by timyerg** [comment]
Thank you for your fast replies and support!
I switched to dev. repo.

Now I switched from line 57 to 59! =)

KeyError in line 38 of /Natrix2/rules/quality_filt.smk:
'threads'
  File "/Natrix2/Snakefile", line 59, in <module>
  File "/Natrix2/rules/quality_filt.smk", line 38, in <module>

So it can not find "threads". Replacing with "cores" did not help.

If I disable it in snakemake file, error changes to:

```
Building DAG of jobs...
MissingInputException in line 198 of /Natrix2/rules/read_correction.smk:
Missing input files for rule fastq2fasta:
Nanopore_results/quality_filtering/C01C0_A_R1.fastq
```
